src/test/datascience/data-viewing/dataViewerDependencyService.unit.test.ts

```diff
@@ -4,118 +4,99 @@
-            version: new SemVer('3.3.3')
-        };
-        pythonExecService = mock<IPythonExecutionService>();
-        installer = mock(ProductInstaller);
         appShell = mock(ApplicationShell);
-        pythonExecFactory = mock(PythonExecutionFactory);
-        interpreterService = mock<IInterpreterService>();
-
-        dependencyService = new DataViewerDependencyService(
-            instance(appShell),
-            instance(installer),
-            instance(pythonExecFactory),
-            instance(interpreterService),
-            false
-        );
+        kernel = instance(mock<IKernel>());
+        dependencyService = new DataViewerDependencyService(instance(appShell), false);
+    });
 
-        when(interpreterService.getActiveInterpreter()).thenResolve(interpreter);
-        when(interpreterService.getActiveInterpreter(anything())).thenResolve(interpreter);
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (instance(pythonExecService) as any).then = undefined;
+    teardown(() => {
+        sinon.restore();
+    });
+
+    test('What if there are no kernel sessions?', async () => {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (pythonExecService as any).then = undefined;
-        when(pythonExecFactory.createActivatedEnvironment(anything())).thenResolve(instance(pythonExecService));
+        (kernel.session as any) = undefined;
+
+        const resultPromise = dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+
+        await assert.isRejected(
+            resultPromise,
+            DataScience.noActiveKernelSession(),
+            'Failed to determine if there was an active kernel session'
+        );
     });
+
     test('All ok, if pandas is installed and version is > 1.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.30.0' });
-        await dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const version = '3.3.3';
+
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
+
+        const result = await dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+        assert.equal(result, undefined);
     });
+
     test('Throw exception if pandas is installed and version is = 0.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.20.0' });
+        const version = '0.20.0';
 
-        const promise = dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
```

Comment:
Same comment as previously

Reply:
I forgot to spy the calls and read what was sent to them -- but, since I agree with you that the imports shouldn't affect the tests, I am just going to revert using Sinon.stubs for the `executeSilently` function.

Reply:
Wait. You shouldn't have to modify the import. This test here doesn't:

https://github.com/microsoft/vscode-jupyter/blob/4af7b2a2fe854871a2733babfe5bf36d506b6f39/src/test/datascience/jupyter/kernels/installationPrompts.vscode.test.ts#L346

Sinon should be able to stub the module as long as you import the module under test AFTER the stubbed module has been imported into the test module.

Reply:
yes, but how am I going to import the module after I stub it if I have the module imported at the top of the file?

Reply:
dataViewerDependencyService.ts = moduleUnderTest 
kernel/helpers.ts = module you need to stub

Top of file:

```
import * as helpers from '.../kernels/helpers.ts'
import { DataViewerDependencyService } from '.../dataviewerDependencyService.ts'

..

suitesetup(() => {
    sinon.stub(helpers, 'executeSilently')
}
```
This is the pattern followed in other locations. helpers is imported before dataviewer dependency so the module you have is the one used by both.

Reply:
Example:
https://sinonjs.org/how-to/stub-dependency/

Reply:
Well, but that is what I have as part of the PR right now (there are a couple of small cleanups that I haven't pushed, but specifically regarding this import), I have: `import * as helpers from '../../../kernels/helpers';` but, as Don said here: https://github.com/microsoft/vscode-jupyter/pull/10545#discussion_r906112779

> Feels like a poor way to get the tests working.
Tomorrow if someone were to change this style of import, then the tests would fail,
i.e. i don't think we should force the import style just to get the tests passing.

Reply:
By changing the imports I believe Don and I refer to the fact that it was `import { executeSilently }` inside of the `dataviewerDependencyService` before I started using Sino.stub

I believe Don's point is that if someone changes that import (as I currently have in this PR on `dataviewerDependencyService`) for something other than `import * as helpers from '../../../kernels/helpers';`, the test will fail.

Please, let me know if I'm missing something.

Reply:
I personally think `import * as helpers from '../../../kernels/helpers';` is ok.

Reply:
You shouldn't need to change the import in DataViewerDependencyService.

Reply:
I'll try!

Reply:
Seems like it's working! I believe this didn't use to work before (like, in the before times), but my memory is really bad.

Reply:
Updated!

Reply:
of you haven't fixed this, why not just stub the kernel.session methods. i think that's much easier.

Reply:
this test look wrong. 
previously we had tests for the method that took an interpreter, now we have lost that test.
i think we need this test in 4 different permutations (local vs remote, debug vs non debug)

Reply:
Nope there's a new test file.